src/v/coproc/wasm_event_listener.cc

```diff
@@ -126,6 +130,38 @@ ss::future<> wasm_event_listener::start() {
       });
 }
 
+ss::future<std::vector<model::record_batch>>
+decompress_wasm_events(model::record_batch_reader::data_t events) {
+    return ssx::parallel_transform(
+      std::move(events), [](model::record_batch rb) {
+          /// If batch isn't compressed, returns 'rb'
+          return storage::internal::decompress_batch(std::move(rb));
+      });
+}
+
+/// Map a model::record_batch -> std::vector<wasm_script_action>
+/// The transform will only occur after the record has passed all validators
+wasm_script_actions
```

Comment:
static. although why not expose this in a `*_utils.h` inside a `wasm::details {}` namespace and add unit tests for these ?